bleak/backends/scanner.py

```diff
@@ -176,7 +185,24 @@ def detection_callback(s, d):
         else:
             detection_callback = callback
 
-        self._callback = detection_callback
+        self._ad_callbacks.append(detection_callback)
```

Comment:
Might want to consider a dict of token: callback instead of list to ensure we don't get unbalanced removals (i.e. it should be safe to call the remove function more than once).